docs/source/cb3.rst

```diff
@@ -0,0 +1,208 @@
+    of the conda docs for more info.
+
+
+.. _compiler-tools:
+
+Compiler tools
+~~~~~~~~~~~~~~
+**Summary:**
+
+- Previously we used ``- gcc #[linux]`` and ``- llvm # [osx]`` for compilers
+- Instead, we should use the syntax ``{{ compiler('c') }}``, ``{{
+  compiler('cxx') }}``, and/or ``{{ compiler('fortran') }}``. These should go
+  in the ``build`` section, and all other build dependencies should go in the
+  ``host`` section.
+
+Anaconda now provides platform-specific compilers that are automatically
+determined. The string ``{{ compiler('c') }}`` will resolve to ``gcc`` on
+Linux, but ``clang`` on macOS. This should greatly simplify recipes, as we no
+longer need to have separate lines for linux and osx.
+
+Note that previously we typically would also add ``- libgcc #[linux]`` as a run
+dependency, but this is now taken care of by the compiler tools.
+
+Conda-build 3 also now has the ability to cross-compile, making it now possible
+to compile packages for macOS while running on Linux. To support this, recipes
+must now make a distinction between dependencies that should be specific to the
+building machine and dependencies that should be specific to the running
+machine.
+
+Dependencies specific to the building machine go in ``build``;
+dependencies specific to the running machine go in ``host`` (see
+:ref:`host-section`).
+
+
+Before:
+
+.. code-block:: yaml
+
+    package:
+      name: example
+      version: 0.1
+    requirements:
+      build:
+        - python
+        - gcc  # [linux]
+        - llvm # [osx]
+      run:
+        - python
+        - libgcc  # [linux]
+
+After:
+
+.. code-block:: yaml
+
+    package:
+      name: example
+      version: 0.1
+    requirements:
+      build:
+        - {{ compiler('c') }}
+      host:
+        - python
+      run:
+        - python
+
+.. seealso::
+
+    - The `compiler tools
+      <https://conda.io/docs/user-guide/tasks/build-packages/compiler-tools.html>_
+      section of the conda docs has much more info.
+
+    - The default compiler options are defined by conda-build in the
+      `variants.DEFAULT_COMPILERS
+      <https://github.com/conda/conda-build/blob/master/conda_build/variants.py#L42>`_
+      variable.
+
+    - More details on "strong" and "weak" exports (using examples of libpng and
+      libgcc) can be found in the `export runtime requirements
+      <https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#export-runtime-requirements>`_
+      conda documentation.
+
+
+.. warning::
+
+    These compilers are only available in the ``anaconda`` channel. Until now
+    we have not had this channel as a dependency, so be sure to add the channel
+    when setting up bioconda (see :ref:`set-up-channels`).
+
+.. _global-pinning:
+
+Global pinning
+~~~~~~~~~~~~~~
+
+**Summary:**
+
+- Previously we pinned packages using the syntax ``- zlib {{ CONDA_ZLIB }}*``
+- Instead, we should now pin packages with the syntax ``- zlib {{ zlib }}``.
```

Comment:
I don't think we should do this. We should wait until zlib and co have run-exports defined and work in the meantime to get ours as well supported for run_exports (https://github.com/bioconda/bioconda-recipes/pull/8831). This would mean don't have to define zlib at all as run-time dependency.

Reply:
After reading the `run_exports` docs some more (and your other comments below), let me completely rewrite those docs!

Reply:
@bgruening just pushed some updates/fixes to this section. Any better?